### PR TITLE
#169566329 Integrate Husky and Lint staged

### PR DESCRIPTION
**Pull request by gildniy**
#### What does this PR do?
It integrates Husky and Lint staged for pre-commit linting and test hook

#### Description of Task to be completed
- [x] Add `husky` and `lint-stagged`
- [x] Create `.husky` with hooks configuration at the project root
- [x] Create the `lint-staged` script in package.json to run and configure it to run `eslint` at any time `git commit` command is called

#### How should this be manually tested?
- By making any link error eg. Putting a `)` at the end of the `src/index.js` file, and run `git commit -m "Test Husky"` in the terminal, an error showing the line having the lint error will show up

#### Any background context you want to provide?
The code shared in a team may be compliant to the linting rules. With Husky and Lint-stagged, the code will be verified and formated before being deploy. 

#### What are the relevant pivotal tracker stories?
[#169566329](https://www.pivotaltracker.com/story/show/169566329)

#### Screenshots (if appropriate)
N/A

### CODE REVIEW

**Review by timolinn — COMMENTED** [review]
Looks great to me. I noticed you have some `idea` files checked in?

**Comment by gildniy** [comment]
> Looks great to me. I noticed you have some `idea` files checked in?

I will remove the `.idea` files, it was a mistake.

**Comment by gildniy** [comment]
> Looks great to me. I noticed you have some `idea` files checked in?

Now fixed.

**Review by timolinn — DISMISSED** [review]

**Review by nshaphan — DISMISSED** [review]

**Review by nshaphan — COMMENTED** [review]
Great but Conflicts need to be resolved

**Review by timolinn — DISMISSED** [review]
LGTM

File: src/models/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+let sequelize;
+if (config.use_env_variable) {
+  sequelize = new Sequelize(process.env[config.use_env_variable]);
+} else {
+  sequelize = new Sequelize(config.database, config.username, config.password, config);
+}
+
+readdirSync(__dirname)
+  .filter((file) => (file.indexOf('.') !== 0) && (file !== basename) && (file.slice(-3) === '.js'))
```
[the remark this answers is not in the document]
Comment (timolinn) [reply src/models/index.js]: @gildniy verify if `.hound.yml` is using the same eslint configuration with barefoot nomad.

**Review by nshaphan — APPROVED** [review]

**Review by nshaphan — DISMISSED** [review]

**Review by b0nbon1 — DISMISSED** [review]
Tested its working okay, Its good to be merged

**Review by RaymondMwaura — COMMENTED** [review]
Maybe you can remove the duplicate 'mocha' dependency under 'devDependencies' since it's already included under 'dependencies' where it should be.

**Comment by gildniy** [comment]
> Maybe you can remove the duplicate 'mocha' dependency under 'devDependencies' since it's already included under 'dependencies' where it should be.

Thanks, now fixed.

**Review by timolinn — APPROVED** [review]

**Review by b0nbon1 — APPROVED** [review]

**Review by nshaphan — APPROVED** [review]

**Pull request merged**